src/gar.lua

```diff
@@ -912,6 +913,8 @@ function gar.updateGatewayStats(address, gateway, stats)
 	return gateway
 end
 
+--- Updates the weights for a gateway
+--- @param weightedGateway WeightedGateway The weighted gateway to update the weights for
 function gar.updateGatewayWeights(weightedGateway)
```

Comment:
Using a copy of the gateway seems pretty clean here, but wondering whether using the unsafe variant of getGateway would also be acceptable...

Reply:
updated in https://github.com/ar-io/ar-io-network-process/pull/353/commits/11b9e1c33bb0975520a93ec5193b074670982a3f